sktm/patchwork.py

```diff
@@ -208,19 +208,46 @@ def get_apiurls(self):
 
         return r.json()
 
-    def get_message_id_and_emails(self, pid):
+    def get_header_value(self, patch_id, key):
         """
-        Get Message-ID and all involved e-mail addresses from patch message
-        headers.
+        Get the value(s) of requested message header.
+
+        Since Patchwork 2.1, all the headers with same key are returned in the
+        API as a list (this is relevant for eg. 'Received' header which can be
+        present multiple times; before, only one of them was present). Work
+        around this difference by concatenating the values with double newlines
+        as a divider (we shouldn't need headers which can be present multiple
+        times anyways).
 
         Args:
-            pid:    ID of the patch to get header values for.
+            patch_id: ID of the patch to retrieve header value for.
+            key:      Key of the header which value should be retrieved.
+
+        Returns:
+            A string representing the value of requested header from patch.
+        """
+        req = requests.get('%s/%s' % (self.apiurls.get('patches'), patch_id))
+
+        if req.status_code != 200:
+            raise Exception('Failed to get data for patch %d (%d)' %
+                            (patch_id, req.status_code))
+
+        header_value = req.json().get('headers', {}).get(key, '')
+        if isinstance(header_value, string):
+            return header_value
+        else:
+            return '\n\n'.join([value for value in header_value])
```

Comment:
This is not normalizing the case of the header name characters, and doesn't account towards case differences between e.g. `Message-Id` and `Message-ID` (yes, I've seen both appear in a single maillist on a Patchwork instance).

Could you please fix that?

Also, since you would need to iterate through the keys anyway here, same as is done in `get_emails()`, do you think you could make the function accept variable number of header name arguments (we wouldn't be able to name them `keys` anymore due to normalization), so that their values could be returned at once (as a tuple), and we could use this function in `get_emails` without loading the message from Patchwork for every header?

Reply:
We would also be able to remove the duplicate request for the subject this way.

Reply:
Valid point. Hm, variable number of arguments sounds neat, I'll get on it!